### PR TITLE
Graphics

### PR DESCRIPTION
**Pull request by elkyhalberstam**

### CODE REVIEW

File: src/main/java/halberstam/weather/CurrentWeatherFrame.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        panel.add(currentWeatherView, BorderLayout.CENTER);
+
+        enterCityButton.addActionListener(e -> {
+            Retrofit retrofit = new Retrofit.Builder()
+                    .baseUrl("https://api.openweathermap.org/")
+                    .addConverterFactory(GsonConverterFactory.create())
+                    .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                    .build();
+
+            OpenWeatherMapService service = retrofit.create(OpenWeatherMapService.class);
```
Comment (andrewoid) [inline src/main/java/halberstam/weather/CurrentWeatherFrame.java]: You should only create the `service` one time. 
Not every time you press the button.

File: src/main/java/halberstam/weather/CurrentWeatherFrame.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+            FiveDayForcast fiveDayForcast =
+                    service.getFiveDayForcast(enterCityName.getText()).blockingFirst();
```
Comment (andrewoid) [inline src/main/java/halberstam/weather/CurrentWeatherFrame.java]: use `subscribe()` not `blockingFirst()`

File: src/main/java/halberstam/weather/CurrentWeatherView.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://api.openweathermap.org/")
+                .addConverterFactory(GsonConverterFactory.create())
+                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                .build();
+
+        OpenWeatherMapService service = retrofit.create(OpenWeatherMapService.class);
+
+        FiveDayForcast fiveDayForcast = service.getFiveDayForcast("Passaic").blockingFirst();
+        this.fiveDayForcast = fiveDayForcast;
```
Comment (andrewoid) [inline src/main/java/halberstam/weather/CurrentWeatherView.java]: You're doing this in the frame.
Don't do it here anymore.

File: src/main/java/halberstam/weather/CurrentWeatherView.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    protected void paintComponent(Graphics g)
+    {
+        super.paintComponent(g);
+        int height = getHeight();
+        g.translate(0, height);
+
+
+
+        java.util.List<List> forcastList = fiveDayForcast.list;
```
Comment (andrewoid) [inline src/main/java/halberstam/weather/CurrentWeatherView.java]: If you remove the code from the constructor, `fiveDayForecast` will be null.
So make sure you don't execute this if `fiveDayForecast` is null

File: src/main/java/halberstam/weather/CurrentWeatherFrame.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    public CurrentWeatherFrame()
+    {
+        setSize(650, 600);
+        setTitle("Current Weather");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        JPanel enterCityPanel = new JPanel();
+        enterCityPanel.setLayout(new FlowLayout());
+
+        JTextField enterCityName  = new JTextField("Passaic             ");
```
Comment (andrewoid) [inline src/main/java/halberstam/weather/CurrentWeatherFrame.java:33]: Can you make it so it automatically requests Passaic when the application starts up?

Comment (elkyhalberstam) [reply src/main/java/halberstam/weather/CurrentWeatherFrame.java:33]: is it better now?

File: src/main/java/halberstam/weather/CurrentWeatherFrame.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl("https://api.openweathermap.org/")
+                .addConverterFactory(GsonConverterFactory.create())
+                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
+                .build();
+
+        OpenWeatherMapService service = retrofit.create(OpenWeatherMapService.class);
+        FiveDayForecast intitalFiveDayForecast =
+                service.getFiveDayForecast(enterCityName.getText()).blockingFirst();
```
Comment (andrewoid) [inline src/main/java/halberstam/weather/CurrentWeatherFrame.java]: BlockingFirst is only allowed in tests

**Review by andrewoid — APPROVED** [review]

**Pull request merged**